fatiando/mesher.py

```diff
@@ -1720,3 +1720,433 @@ def keep(cell):
+                     'remanence':[10000., 25., 40.],
+                     'k':[0.562, 0.485, 0.25, 90., 0., 0.]}``
+
+    Examples:
+
+        >>> e = TriaxialEllipsoid(1, 2, 3, 6, 5, 4, 10, 20, 30,{
+        ...                       'remanence':[10000., 25., 40.],
+        ...                       'k': [0.562, 0.485, 0.25, 90., 0., 0.]})
+        >>> e.props['remanence']
+        [10000.0, 25.0, 40.0]
+        >>> e.addprop('density', 20)
+        >>> print e.props['density']
+        20
+        >>> print e
+        x:1.0 | y:2.0 | z:3.0 | a:6.0 | b:5.0 | c:4.0 | alpha:10.0 | gamma:20.\
+0 | delta:30.0 | density:20 | k:[0.562, 0.485, 0.25, 90.0, 0.0, 0.0] | remanen\
+ce:[10000.0, 25.0, 40.0]
+        >>> e = TriaxialEllipsoid(1, 2, 3, 6, 5, 4, 10, 20, 30)
+        >>> print e
+        x:1.0 | y:2.0 | z:3.0 | a:6.0 | b:5.0 | c:4.0 | alpha:10.0 | gamma:20.\
+0 | delta:30.0
+        >>> e.addprop('density', 2670)
+        >>> print e
+        x:1.0 | y:2.0 | z:3.0 | a:6.0 | b:5.0 | c:4.0 | alpha:10.0 | gamma:20.\
+0 | delta:30.0 | density:2670
+
+    """
+
+    def __init__(self, x, y, z, a, b, c, alpha, gamma, delta, props=None):
+        GeometricElement.__init__(self, props)
+
+        self.x = float(x)
+        self.y = float(y)
+        self.z = float(z)
+        self.a = float(a)
+        self.b = float(b)
+        self.c = float(c)
+        self.alpha = float(alpha)
+        self.gamma = float(gamma)
+        self.delta = float(delta)
+
+        assert self.a > self.b and self.b > self.c, \
+            "a must be greater than b and b must greater than c"
+
+    def __str__(self):
+        """
+        Return a string representation of the ellipsoid.
+        """
+
+        names = [('x', self.x), ('y', self.y), ('z', self.z),
+                 ('a', self.a), ('b', self.b), ('c', self.c),
+                 ('alpha', self.alpha), ('gamma', self.gamma),
+                 ('delta', self.delta)]
+        names = names + [(p, self.props[p]) for p in sorted(self.props)]
+        return ' | '.join('%s:%s' % (n, v) for n, v in names)
+
+    def V(self, angles=None):
+        '''
+        Calculates the coordinate transformation matrix
+        for a triaxial model. The columns of this matrix
+        are defined according to the unit vectors v1, v2
+        and v3 presented by Clark et al. (1986, p. 192).
+
+        Parameters:
+
+        * angles: array like containing the three
+            orientation angles (alpha, gamma, delta) of the main
+            susceptibility axes.
+
+        Returns:
+
+        *matrix: numpy array 2D
+                 A 3x3 matrix with the direction cosines of the new coordinate
+                 system.
+
+        References:
+
+        Clark, D., Saul, S., and Emerson, D.: Magnetic and gravity anomalies
+        of a triaxial ellipsoid, Exploration Geophysics, 17, 189-200, 1986.
+
+        Example:
+
+            >>> e = TriaxialEllipsoid(11, 3, 4, 3, 2, 1, 180, 90, 0)
+            >>> print e.V()
```

Comment:
Don't print numpy arrays on doctests. This will break eventually for some Windows/Python version because of floating point precision. This test belongs on a test file and checking using something like `numpy.testing.assert_allclose`.